File: problems/java/07. Reverse Integer.java
class Solution {
    public int reverse(int x) {
        int rev = 0;
        while (x != 0) {
            int dig = x % 10;
            x /= 10;

            if (rev > Integer.MAX_VALUE / 10 || (rev == Integer.MAX_VALUE / 10 && dig > 7)) {
                return 0;
            }
            if (rev < Integer.MIN_VALUE / 10 || (rev == Integer.MIN_VALUE / 10 && dig < -8)) {
                return 0;
            }
            
            rev = rev * 10 + dig;
        }
        return rev;
    }
}

File: problems/Rust/07. Reverse Integer.rs
impl Solution {
    pub fn reverse(x: i32) -> i32 {
        let mut rev = 0;
        let mut x = x;

        while x != 0 {
            let dig = x % 10;
            x /= 10;

            if rev > i32::MAX / 10 || (rev == i32::MAX / 10 && dig > 7) {
                return 0;
            }
            if rev < i32::MIN / 10 || (rev == i32::MIN / 10 && dig < -8) {
                return 0;
            }
            
            rev = rev * 10 + dig;
        }
        rev
    }
}